llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -813,14 +813,29 @@ Instruction *InstCombinerImpl::foldGEPICmp(GEPOperator *GEPLHS, Value *RHS,
       }
     }
 
-    // Only lower this if the icmp is the only user of the GEP or if we expect
-    // the result to fold to a constant!
-    if ((GEPsInBounds || CmpInst::isEquality(Cond)) &&
-        (GEPLHS->hasAllConstantIndices() || GEPLHS->hasOneUse()) &&
-        (GEPRHS->hasAllConstantIndices() || GEPRHS->hasOneUse())) {
+    if (GEPsInBounds || CmpInst::isEquality(Cond)) {
+      auto EmitGEPOffsetAndRewrite = [&](GEPOperator *GEP) {
+        auto *Inst = dyn_cast<Instruction>(GEP);
```

Comment:
IIRC we have removed constant expr support for `getelementptr`. I cannot find the pattern `(getelementptr` in the LLVM regression tests and my benchmarks.

If not, should we restore the insert point after offset expansion?

Reply:
I've added an insert pointer guard, though I think it shouldn't make a difference in practice.

Getelementptr constant expressions exist, but the syntax is `getelementptr (`, not `(getelementptr`.